tooling/build/scripts/publishing/queries.ts

```diff
@@ -0,0 +1,55 @@
+export const GET_ALL_RESOURCES_WITH_FULL_PERMALINKS = `
+    WITH RECURSIVE "resourcePath" (id, title, permalink, parentId, type, content, "fullPermalink", "publishedVersionId") AS (
+    -- Base case for all resources
+    SELECT
+        r.id,
+        r.title,
+        r.permalink,
+        r."parentId",
+        r.type,
+        CASE WHEN r.type = 'Page' THEN b."content" ELSE NULL END AS content,
+        r.permalink AS "fullPermalink",
+        r."publishedVersionId"
+    FROM
+        public."Resource" r
+    LEFT JOIN public."Version" v ON v."id" = r."publishedVersionId"
+    LEFT JOIN public."Blob" b ON v."blobId" = b.id
+    WHERE
+        r."siteId" = $1 AND r."parentId" IS NULL
+
+    UNION ALL
+
+    -- Recursive case
+    SELECT
+        r.id,
+        r.title,
+        r.permalink,
+        r."parentId",
+        r.type,
+        CASE WHEN r.type = 'Page' THEN b."content" ELSE NULL END AS content,
+        CONCAT(path."fullPermalink", '/', r.permalink) AS "fullPermalink",
+        r."publishedVersionId"
+    FROM
+        public."Resource" r
+        LEFT JOIN public."Version" v ON v."id" = r."publishedVersionId"
+        LEFT JOIN public."Blob" b ON v."blobId" = b.id
+
+        -- This join determines if the recursion continues if there are more rows
+        INNER JOIN "resourcePath" path ON r."parentId" = path.id
+    WHERE
+        r."siteId" = $1 AND (r.type = 'Folder' OR (r.type = 'Page' AND b."content" IS NOT NULL))
+    )
+    SELECT * FROM "resourcePath";
```

Comment:
issue: I think this is retrieving all the pages of the site but instead it should be only retrieving pages with a `publishedVersionId`.

Reply:
@dcshzj the left join is on the publishedVersionId right

```
LEFT JOIN public."Version" v ON v."id" = r."publishedVersionId"
```

Reply:
Hmm yes but left join means that all rows on the left table (Resources) [would be kept](https://stackoverflow.com/a/51619897), even if the `on` condition with the NULL values is evaluated to false.

Reply:
Ah i see what you mean, will fix!

Reply:
I have added a where clause at the end now